src/Catalog/Monitoring/Validation/Test/RegistrationValidationTests.cs

```diff
@@ -0,0 +1,138 @@
+                : base(timestampMetadataResourceV2)
+            {
+                _v2Metadata = v2Metadata.Resource ??
+                    throw new ArgumentNullException(nameof(v2Metadata));
+                _v3Metadata = v3Metadata.Resource ??
+                    throw new ArgumentNullException(nameof(v3Metadata));
+                _regResource = regResource.Resource ??
+                    throw new ArgumentNullException(nameof(regResource));
+            }
+
+            private readonly MetadataResource _v2Metadata;
+            private readonly MetadataResource _v3Metadata;
+
+            private readonly RegistrationResourceV3 _regResource;
+
+            public override async Task RunInternal(ValidationTestRunData data)
+            {
+                var v2PackageIdExists = await _v2Metadata.Exists(data.Package.Id, true, true, NullLogger.Instance, data.CancellationToken);
+                var v3PackageIdExists = await _v3Metadata.Exists(data.Package.Id, true, true, NullLogger.Instance, data.CancellationToken);
+
+                Assert.Equal(v2PackageIdExists, v3PackageIdExists);
+            }
+        }
+
+        /// <summary>
+        /// Verifies that the leaf node blob for the package exists when it should.
+        /// </summary>
+        public class RegistrationLeafExistsValidationTest : ValidationTestRunner, IValidationTestRunner<RegistrationEndpoint>
+        {
+            public RegistrationLeafExistsValidationTest(
+                ResourceToTest<IPackageTimestampMetadataResource, V2ResourceType> timestampMetadataResourceV2,
+                ResourceToTest<MetadataResource, V2ResourceType> v2Metadata,
+                ResourceToTest<MetadataResource, V3ResourceType> v3Metadata,
+                ResourceToTest<RegistrationResourceV3, V3ResourceType> regResource)
+                : base(timestampMetadataResourceV2)
+            {
+                _v2Metadata = v2Metadata.Resource ??
+                    throw new ArgumentNullException(nameof(v2Metadata));
+                _v3Metadata = v3Metadata.Resource ??
+                    throw new ArgumentNullException(nameof(v3Metadata));
+                _regResource = regResource.Resource ??
+                    throw new ArgumentNullException(nameof(regResource));
+            }
+
+            private readonly MetadataResource _v2Metadata;
+            private readonly MetadataResource _v3Metadata;
+
+            private readonly RegistrationResourceV3 _regResource;
+
+            public override async Task RunInternal(ValidationTestRunData data)
+            {
+                var v2IndividualPackageExists = await _v2Metadata.Exists(data.Package, NullLogger.Instance, data.CancellationToken);
+                var v3IndividualPackageExists = await _v3Metadata.Exists(data.Package, NullLogger.Instance, data.CancellationToken);
+
+                if (v2IndividualPackageExists != v3IndividualPackageExists)
+                {
+
```

Comment:
I think so--must have forgotten to finish this code. This PR is for the infrastructure as a whole and not the individual validations so you can feel free to ignore it. Apologies!